src/azure-cli/azure/cli/command_modules/synapse/operations/workspace.py

```diff
@@ -41,6 +44,28 @@ def create_workspace(cmd, client, resource_group_name, workspace_name, storage_a
         else:
             managed_virtual_network_settings = ManagedVirtualNetworkSettings(prevent_data_exfiltration=False)
 
+    if repository_type:
```

Comment:
What if customer provide all other details without specifying repository type? Should we add a warning to tell customer that all others will be ignored without `--repository-type`?

Reply:
Yes, will add a warning. Thanks.